wordpress-mobile/WordPress-iOS
# The themes feature can now be enabled with a URL.

Closes [#3900](https://github.com/wordpress-mobile/WordPress-iOS/issues/3900) and [3901](https://github.com/wordpress-mobile/WordPress-iOS/issues/3901).

Implements a special URL through which the themes feature can be turned ON or OFF.  Turning the feature ON or OFF means the feature will be hidden or visible to the user when they open a blog's details screen.

The themes menu is still not wired to do anything.  It's also hidden for all non-debug builds.

URL to enable themes: `wpdebug://themes?enabled=1`
URL to disable themes: `wpdebug://themes?enabled=0`

**Test 1:**
1. Open the app.
2. Go to a blog's details screen.
3. Make sure the themes menu is hidden by default.
4. Make sure all options from the menu work regularly.

**Test 2:**
1. Open the app, make sure you're not inside a blog's detail screen
2. Without closing the app go to mobile safari.
3. Paste the URL: `wpdebug://themes?enabled=1`
4. The app should open and you should see a success message.
5. Go to a blog's details screen.  Make sure the "Themes" menu is visible.

**Test 3:**
1. Enable themes as in test 2.
2. Go to mobile safari and paste the URL: `wpdebug://themes?enabled=0`
3. The app should open and you should see a success message (telling you the feature is disabled).
4. Go to a blog's details screen and make sure the themes option is not visible.

**Test 4:**
1. Enable or disable the themes feature while the blog details screen is open.
2. The feature will not be updated in real time, but make sure the menu options are NOT broken.
